lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -919,7 +929,21 @@ void HdVP2Mesh::Sync(
                 MProfiler::kColorC_L2,
                 _rprimId.asChar(),
                 "HdVP2Mesh::GetMeshTopology");
-            _meshSharedData->_topology = GetMeshTopology(delegate);
+            HdMeshTopology newTopology = GetMeshTopology(delegate);
+
+            // Test to see if the topology actually changed. If not, we don't have to do anything!
+            // Don't test IsTopologyDirty anywhere below this because it is not accurate. Instead
+            // using the _indexBufferValid flag on render item data.
```

Comment:
I need the separate flag because we've already lost dirty information by the time sync is called. In propagate dirty bits the old topology dirty flag saved the the render item was overwritten. If that item hasn't been getting updated (because it's in a repr that is not currently displayed) then we don't know if there was a ligitimate topology change before this frame that still needs to be handled for that item.

For that reason I need the separate _indexBufferValid flag per render item.